### PR TITLE
Correctly convert literal protos to Literals.

### PR DESCRIPTION
**Pull request by andreweland**
Correct a bug where literal protos would always return an Expression with a nil AnyExpression. Also return Literal instead of Expression, and correctly fill in the name and token positions.

### CODE REVIEW

File: src/diagonal.works/b6/expression.go [hunk 1]
```diff
@@ -228,15 +232,15 @@ func (l Literal) ToProto() (*pb.NodeProto, error) {
 	return l.AnyLiteral.ToProto()
 }
 
-func LiteralFromProto(node *pb.NodeProto) (Expression, error) {
-	var e Expression
```
Comment (andreweland) [inline src/diagonal.works/b6/expression.go:232]: This was the crux of the bug. e is defined here, but immediately shadowed below.

Comment (xprogramari) [reply src/diagonal.works/b6/expression.go:232]: UGH tyyyyy

Comment (xprogramari) [reply src/diagonal.works/b6/expression.go:232]: u didnt submit this

Comment (andreweland) [reply src/diagonal.works/b6/expression.go:237]: Sorry, I've been a bit preoccupied and forgot about it! Rebasing and submitting now.

**Pull request merged**